common/client-libs/validator-client/src/nymd/mod.rs

```diff
@@ -88,7 +88,7 @@ impl NymdClient<SigningNymdClient> {
         vesting_contract_address: Option<AccountId>,
         erc20_bridge_contract_address: Option<AccountId>,
         signer: DirectSecp256k1HdWallet,
-        gas_price: Option<GasPrice>,
+        gas_price: GasPrice,
```

Comment:
So `connect_with_signer` makes `gas_price` mandatory, while in `connect_with_mnemonic` it's still optional -> any particular reason for that choice?

Reply:
Also with so many arguments, maybe at some point there should be a `Config` struct passed to the client? (Not unneccesarily now, but perhaps at some point) I guess it could have fields like all the contract addresses, gas price, etc.

Reply:
> So `connect_with_signer` makes `gas_price` mandatory, while in `connect_with_mnemonic` it's still optional -> any particular reason for that choice?

In `connect_with_mnemonic` we have access to `denom` so we can still use the default gas price in case it's not provided. Not so in `connect_with_signer`